.husky/pre-commit

```
@@ -1,5 +1,8 @@
+# undo staged changes to blog data
+git restore --staged apps/site/public/blog-data.json
```

Can we do this in a follow up? I'm -1 on using pre commit to unstage a file, and I think that should be a separate discussion

---

agreed

---

If no one has strong feelings, it shouldn't. This is what I originally proposed to @dario-piotrowicz that I wanted to do to prevent builds to fail without blog generation or requiring a pre-step that is out of the control of the next.js's process.

If anyone has strong feelings, then sure. We can remove this

---

I have strong feelings. I feel like this will lead to a lot of contributors accidentally including this file in their work, so I'd like to a separate discussion regarding it.

Using the ignore file, we can ignore without requiring the user to use husky to commit (which, while ideal, not everyone may)

---

Both of our current approaches have flaws. Using an ignore file breaks tests and builds if the necessary pre-script hasn't been run for every required command. For instance, if I update a test that depends on this pre-script and haven't executed it, the entire process fails.

Using an initial value at least ensures that the project builds and tests can run. However, one could argue that it's better for these situations to fail explicitly rather than pass silently.

Ultimately, I believe this external data should be generated during the Next.js build or runtime, though I'm unsure if Next.js natively supports this. The core issue is that we're generating an external field that Next.js expects to always be available.

The current setup is also problematic for local development, as the blog-data only updates after a full restart. Ideally, we should refactor this to eliminate the need for an external script altogether.

---

I think we should take this out and discuss in a separate PR/issue, there is clearly a conversation to be had.

What's the intent here? If its just to have `blog-data.json` exist as an empty file by default, and ensure that local changes don't get checked in, why can't we check in the empty file (as this PR does) and then git ignore it (instead of this revert cmd) so that future local changes aren't tracked?

---

> What's the intent here? If its just to have blog-data.json exist as an empty file by default, and ensure that local changes don't get checked in, why can't we check in the empty file (as this PR does) and then git ignore it (instead of this revert cmd) so that future local changes aren't tracked?

@MattIPv4 yes that's the intent and your solution would be ideal, unfortunately I think that that is not how git works, git either tracks a file or doesn't track it, it can't track a file to a certain point and ignore any future revisions of it

see: https://github.com/nodejs/nodejs.org/pull/7383#discussion_r2025570672 and https://github.com/nodejs/nodejs.org/pull/7383#discussion_r2027468571

---

This is still the thing that bugs me, because I do _think_ that git allows to ignore a file after it got commited and won't track future changes, let me try that lol.

---

I've addressed the comments.